frame/plasm-rewards/src/lib.rs

```diff
@@ -0,0 +1,428 @@
+    pub enum Error for Module<T: Trait> {
+        /// Duplicate index.
+        DuplicateIndex,
+        /// Invalid era to reward.
+        InvalidEraToReward,
+    }
+}
+
+decl_module! {
+    pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+        /// Number of sessions per era.
+        const SessionsPerEra: SessionIndex = T::SessionsPerEra::get();
+
+        type Error = Error<T>;
+
+        fn deposit_event() = default;
+
+        fn on_runtime_upgrade() {
+            migrate::<T>();
+        }
+
+        /// On finalize is called at after rotate session.
+        fn on_finalize() {
+            // Set the start of the first era.
+            if let Some(mut active_era) = Self::active_era() {
+                if active_era.start.is_none() {
+                    active_era.start = Some(T::Time::now());
+                    <ActiveEra<T>>::put(active_era);
+                }
+            }
+        }
+
+        // ----- Root calls.
+        /// Force there to be no new eras indefinitely.
+        ///
+        /// # <weight>
+        /// - No arguments.
+        /// # </weight>
+        #[weight = SimpleDispatchInfo::FixedNormal(5_000)]
+        fn force_no_eras(origin) {
+            ensure_root(origin)?;
+            ForceEra::put(Forcing::ForceNone);
+        }
+
+        /// Force there to be a new era at the end of the next session. After this, it will be
+        /// reset to normal (non-forced) behaviour.
+        ///
+        /// # <weight>
+        /// - No arguments.
+        /// # </weight>
+        #[weight = SimpleDispatchInfo::FixedNormal(5_000)]
+        fn force_new_era(origin) {
+            ensure_root(origin)?;
+            ForceEra::put(Forcing::ForceNew);
+        }
+
+        /// Force there to be a new era at the end of sessions indefinitely.
+        ///
+        /// # <weight>
+        /// - One storage write
+        /// # </weight>
+        #[weight = SimpleDispatchInfo::FixedNormal(5_000)]
+        fn force_new_era_always(origin) {
+            ensure_root(origin)?;
+            ForceEra::put(Forcing::ForceAlways);
+        }
+
+        /// Set history_depth value.
+        ///
+        /// Origin must be root.
+        #[weight = SimpleDispatchInfo::FixedOperational(500_000)]
+        fn set_history_depth(origin, #[compact] new_history_depth: EraIndex) {
+            ensure_root(origin)?;
+            if let Some(current_era) = Self::current_era() {
+                HistoryDepth::mutate(|history_depth| {
+                    let last_kept = current_era.checked_sub(*history_depth).unwrap_or(0);
+                    let new_last_kept = current_era.checked_sub(new_history_depth).unwrap_or(0);
+                    for era_index in last_kept..new_last_kept {
+                        Self::clear_era_information(era_index);
+                    }
+                    *history_depth = new_history_depth
+                })
+            }
+        }
+    }
+}
+
+fn migrate<T: Trait>() {
```

Comment:
What's the reason for using it?

Reply:
The process inside is currently a meaningless process. However, the processing of the runtime upgrade will continue to be necessary, so we will consider leaving the implementation empty.